pkg/controller/cluster/cluster_controller.go

```diff
@@ -356,7 +366,91 @@ func (c *Controller) processNextWorkItem() bool {
 	return true
 }
 
+func (c *Controller) buildRemoteClusterClient(cluster *clusteroperator.Cluster) (clusterapiclient.Interface, error) {
+
+	secretName := cluster.Name + "-kubeconfig"
+
+	secret, err := c.kubeClient.CoreV1().Secrets(cluster.Namespace).Get(secretName, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	secretData := secret.Data["kubeconfig"]
+
+	config, err := clientcmd.Load(secretData)
+	if err != nil {
+		return nil, err
+	}
+
+	kubeConfig := clientcmd.NewDefaultClientConfig(*config, &clientcmd.ConfigOverrides{})
+	restConfig, err := kubeConfig.ClientConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	clusterAPIClient, err := clusterapiclient.NewForConfig(restConfig)
+	if err != nil {
+		return nil, err
+	}
+	return clusterAPIClient, nil
+}
+
+func (c *Controller) ensureRemoteMachineSetsAreDeleted(cluster *clusteroperator.Cluster) error {
+	clusterAPIClient, err := c.buildRemoteClusterClient(cluster)
+	if err != nil {
+		return err
+	}
+	machineSets, err := clusterAPIClient.ClusterV1alpha1().MachineSets(remoteClusterNamespace).List(metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	// If all remote machinesets are gone, then we should update the cluster status
+	if len(machineSets.Items) == 0 {
+		original := cluster
+		cluster = cluster.DeepCopy()
+		cluster.Status.DeprovisionedComputeMachinesets = true
+		return controller.PatchClusterStatus(c.client, original, cluster)
+	}
+
+	for _, ms := range machineSets.Items {
+		if ms.DeletionTimestamp != nil {
+			err := clusterAPIClient.ClusterV1alpha1().MachineSets(remoteClusterNamespace).Delete(ms.Name, &metav1.DeleteOptions{})
```

Comment:
Can we use `DeleteCollection` to delete all the relevant machine sets with one call rather than making separate `Delete` calls?

Reply:
Yup, will fix